.generator/templates/response.mustache

```diff
@@ -0,0 +1,233 @@
+	pg = newPaginationInHeader(r)
+	response := &APIResponse{Response: r, cli: cli, pg: pg}
+	return response 
+}
+
+func buildResponse(resp *http.Response,cli *APIClient, v interface{}) (*APIResponse, error) {
+	ct := resp.Header.Get("Content-Type")
+	response := newAPIResponse(resp, cli, v)
+	// err := CheckResponseForError(resp)
+	// if err != nil {
+	// 	return response, err
+	// }
+	var err error
+	bodyBytes, _ := ioutil.ReadAll(resp.Body)
+	copyBodyBytes := make([]byte, len(bodyBytes))
+	copy(copyBodyBytes, bodyBytes)
+	_ = resp.Body.Close()                                    // close it to avoid memory leaks
+	resp.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes)) // restore the original response body
+	if len(copyBodyBytes) == 0 {
+		return response, nil
+	}
+	switch {
+	case strings.Contains(ct, "application/xml"):
+		err = xml.NewDecoder(bytes.NewReader(copyBodyBytes)).Decode(v)
+	case strings.Contains(ct, "application/json"):
+		err = json.NewDecoder(bytes.NewReader(copyBodyBytes)).Decode(v)
+	case strings.Contains(ct, "application/octet-stream"):
+		// since the response is arbitrary binary data, we leave it to the user to decode it
+		return response, nil
+	default:
+		return nil, errors.New("could not build a response for type: " + ct)
+	}
+	if err == io.EOF {
+		err = nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	return response, nil
+}
+
+// TODO check to use model_error
+// func CheckResponseForError(resp *http.Response) error {
+// 	statusCode := resp.StatusCode
+// 	if statusCode >= http.StatusOK && statusCode < http.StatusBadRequest {
+// 		return nil
+// 	}
+// 	e := Error{}
+// 	if (statusCode == http.StatusUnauthorized || statusCode == http.StatusForbidden) &&
+// 		strings.Contains(resp.Header.Get("Www-Authenticate"), "Bearer") {
+// 		for _, v := range strings.Split(resp.Header.Get("Www-Authenticate"), ", ") {
+// 			if strings.Contains(v, "error_description") {
+// 				_, err := toml.Decode(v, &e)
+// 				if err != nil {
+// 					message := "unauthorized"
+// 					e.ErrorSummary = &message
+// 				}
+// 				return &e
+// 			}
+// 		}
+// 	}
+// 	bodyBytes, _ := ioutil.ReadAll(resp.Body)
+// 	copyBodyBytes := make([]byte, len(bodyBytes))
+// 	copy(copyBodyBytes, bodyBytes)
+// 	_ = resp.Body.Close()
+// 	resp.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+//  _ = json.NewDecoder(bytes.NewReader(copyBodyBytes)).Decode(&e)
+// 	if statusCode == http.StatusInternalServerError {
+// 		e.ErrorSummary += fmt.Sprintf(", x-okta-request-id=%s", resp.Header.Get("x-okta-request-id"))
+// 	}
+// 	return &e
+// }
+
+func (res *APIResponse) Next(v interface{}) (*APIResponse, error) {
+	if res.cli == nil {
+		return nil, errors.New("no initial response provided from previous request")
+	}
+	req, err := res.cli.NewRequest(http.MethodGet, res.NextPage(), nil)
+		if err != nil {
+		return nil, err
+	}
+	return res.cli.do(res.cli.cfg.Context, req, v)
+}
+
+// func (res *APIResponse) Next(v interface{}) (*APIResponse, error) {
```

Comment:
@monde can you run the generator and take a look at the code? The code should work fine but I am not sure which approach I should take